lib/src/dart_download.dart

```diff
@@ -0,0 +1,250 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_downloader/flutter_downloader.dart';
+
+/// Factory interface
+typedef CustomHttpClientFactory = HttpClient Function();
+
+// ignore_for_file: use_if_null_to_convert_nulls_to_bools
```

Comment:
Let's follow this linter rule :)

Reply:
Really? Out of my view that makes the code much less readable, ignoring this lint rule results in better readable code out of my view. But since you are the maintainer you have the choice :-)